sql/src/main/java/org/apache/druid/sql/calcite/schema/SystemSchema.java

```diff
@@ -736,37 +733,48 @@ public void close()
   //Note that overlord must be up to get tasks
   private static JsonParserIterator<TaskStatusPlus> getTasks(
       DruidLeaderClient indexingServiceClient,
-      ObjectMapper jsonMapper,
-      BytesAccumulatingResponseHandler responseHandler
+      ObjectMapper jsonMapper
   )
   {
     Request request;
+    InputStreamFullResponseHolder responseHolder;
     try {
+      String query = "/druid/indexer/v1/tasks";
+
       request = indexingServiceClient.makeRequest(
           HttpMethod.GET,
-          "/druid/indexer/v1/tasks",
-          false
+          query
+      );
+
+      responseHolder = indexingServiceClient.go(
+          request,
+          new InputStreamFullResponseHandler()
       );
+
+      if (responseHolder.getStatus().getCode() != HttpServletResponse.SC_OK) {
+        throw new RE(
+            "Failed to talk to overlord leader at [%s]. Error code[%d], description[%s].",
+            query,
```

Comment:
same comment about replacing `query` with more meaningful info here and at other places

Reply:
same